js/renovation/ui/grids/data_grid_next/scrolling/virtual_content.tsx

```diff
@@ -0,0 +1,104 @@
+import {
+  Component, JSXComponent, ComponentBindings,
+  Fragment, InternalState, Consumer, Effect,
+} from '@devextreme-generator/declarations';
+import {
+  Plugins, PluginsContext,
+} from '../../../../utils/plugin/context';
+import { PlaceholderExtender } from '../../../../utils/plugin/placeholder_extender';
+import { TopRowPlaceholder, BottomRowPlaceholder } from '../views/table_content';
+import {
+  TopVirtualRowHeightValue, BottomVirtualRowHeightValue,
+  CalculateTopVirtualRowHeight, CalculateBottomVirtualRowHeight,
+} from './plugins';
+import { VirtualRow } from './virtual_row';
+import {
+  VisibleColumns,
+} from '../plugins';
+import { GetterExtender } from '../../../../utils/plugin/getter_extender';
+
+export const viewFunction = ({
+  topHeight, bottomHeight, cellClasses,
+}: VirtualContent): JSX.Element => (
+  <Fragment>
+    <GetterExtender
+      type={TopVirtualRowHeightValue}
+      order={0}
+      value={CalculateTopVirtualRowHeight}
+    />
+    <GetterExtender
+      type={BottomVirtualRowHeightValue}
+      order={0}
+      value={CalculateBottomVirtualRowHeight}
+    />
+    { topHeight > 0 && (
+      <PlaceholderExtender
+        type={TopRowPlaceholder}
+        order={1}
+        template={(): JSX.Element => (
+          <VirtualRow
+            height={topHeight}
+            cellClasses={cellClasses}
+            rowKey={-1}
```

Comment:
no need in react key here, cause Virtual Row is not in elements array

Reply:
This key is required to render virtual rows among data rows